### DISCUSSION TITLE
issue: Unexpected blur when Controller is wrapped in another component

### DISCUSSION THREAD

**Discussion by echentw** [react-hook-form/react-hook-form#10224] [Q&A]
### Version Number

7

### Codesandbox/Expo snack

https://codesandbox.io/s/hardcore-cerf-uwk4le?file=/src/App.js

### Steps to reproduce

1. Go to https://codesandbox.io/s/hardcore-cerf-uwk4le?file=/src/App.js
2. Edit the text field for the first time.
3. After the first edit, the text field is blurred.

### Expected behaviour

The text field should be focused.

Relevant context:
- This bug happens only if the `isDirty` property is accessed.
- If you remove the `Field` indirection (i.e. inline the `Controller`), then the bug also doesn't happen.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
Your input component is getting mount and remounted on each re-render.

**Reply by echentw** [reply]
Is it possible to nest `<Controller/>` elements like this and avoid the unexpected blur behavior? Possibly with a `ref` or something?
